pkg/controllers/provisioning/scheduling/topology_test.go

```diff
@@ -1125,6 +1126,544 @@ var _ = Describe("Topology", func() {
+			}, count)
+			pods = append(pods, test.UnschedulablePods(test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: lo.Assign(labels, map[string]string{matchedLabel: "value-b"}),
+				},
+				TopologySpreadConstraints: topology,
+			}, count)...)
+			ExpectApplied(ctx, env.Client, nodePool)
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pods...)
+			ExpectSkew(ctx, env.Client, "default", &topology[0]).To(ConsistOf(2, 2))
+		})
+
+		It("should ignore unknown labels specified in matchLabelKeys", func() {
+			matchedLabel := "test-label"
+			topology := []corev1.TopologySpreadConstraint{{
+				TopologyKey:       corev1.LabelHostname,
+				WhenUnsatisfiable: corev1.DoNotSchedule,
+				LabelSelector:     &metav1.LabelSelector{MatchLabels: labels},
+				MatchLabelKeys:    []string{matchedLabel},
+				MaxSkew:           1,
+			}}
+			pods := test.UnschedulablePods(test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				TopologySpreadConstraints: topology,
+			}, 4)
+			ExpectApplied(ctx, env.Client, nodePool)
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pods...)
+			ExpectSkew(ctx, env.Client, "default", &topology[0]).To(ConsistOf(1, 1, 1, 1))
+		})
+	})
+
+	Context("NodeTaintsPolicy", func() {
+		BeforeEach(func() {
+			if env.Version.Minor() < 26 {
+				Skip("NodeTaintsPolicy only enabled by default for K8s >= 1.26.x")
+			}
+		})
+
+		It("should balance pods across a label (NodeTaintsPolicy=ignore)", func() {
+
+			const spreadLabel = "fake-label"
+			nodePool.Spec.Template.Labels = map[string]string{
+				spreadLabel: "baz",
+			}
+			node1 := test.Node(test.NodeOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						spreadLabel: "foo",
+					},
+				},
+				Taints: []corev1.Taint{
+					{
+						Key:    "taintname",
+						Value:  "taintvalue",
+						Effect: corev1.TaintEffectNoSchedule,
+					},
+				},
+				Allocatable: corev1.ResourceList{
+					corev1.ResourceCPU: resource.MustParse("100m"),
+				}})
+
+			node2 := test.Node(test.NodeOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						spreadLabel: "bar",
+					},
+				},
+				Taints: []corev1.Taint{
+					{
+						Key:    "taintname",
+						Value:  "taintvalue",
+						Effect: corev1.TaintEffectNoSchedule,
+					},
+				},
+				Allocatable: corev1.ResourceList{
+					corev1.ResourceCPU: resource.MustParse("100m"),
+				}})
+			ExpectApplied(ctx, env.Client, nodePool, node1, node2)
+
+			ExpectReconcileSucceeded(ctx, nodeStateController, client.ObjectKeyFromObject(node1))
+			ExpectReconcileSucceeded(ctx, nodeStateController, client.ObjectKeyFromObject(node2))
+
+			// there are now three domains for spreadLabel that Karpenter should know about, foo/bar from the two existing
+			// nodes and baz from the node pool
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov)
+			ignore := corev1.NodeInclusionPolicyIgnore
```

Comment:
nit: Do you need to set this to a variable -- can't you just take the address of the string from upstream?